src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabProjectSCMEvent.java

```diff
@@ -54,13 +66,65 @@ public String getSourceName() {
 
     @Override
     public boolean isMatch(@NonNull SCMNavigator navigator) {
-        return navigator instanceof GitLabSCMNavigator && ((GitLabSCMNavigator) navigator).getNavigatorProjects().contains(getPayload().getPathWithNamespace());
+        return navigator instanceof GitLabSCMNavigator && isMatch((GitLabSCMNavigator) navigator);
+    }
+
+    private boolean isMatch(@Nonnull GitLabSCMNavigator navigator) {
+        switch (getType()) {
+            case CREATED:
+                String projectPathWithNamespace = getPayload().getPathWithNamespace();
+                GitLabApi gitLabApi = null;
+                try {
+                    gitLabApi = GitLabHelper.apiBuilder(navigator.getServerName());
+                    GitLabOwner gitlabOwner = GitLabOwner.fetchOwner(gitLabApi, navigator.getProjectOwner());
```

Comment:
This seems wrong. You shouldn't be calling out to a remote system to determine if the payload matches

Reply:
Thanks for the thorough review, I knew it looked iffy but wasn't sure 😓 hence me pinging you. Again, thank you very much ❤️